### ISSUE TITLE
Missing Linting/Validation

### ISSUE THREAD

**Issue by nik-swarmagency** [sachinraja/rodemirror#26]
First of all, absolutely love the work you've done. It's rare to see a good lean wrapper that just works. So thanks for your work.

I was wondering if it would be possible to add linting and validation to this? Similar to: https://codemirror.net/demo/lint.html

Thanks

**Comment by sachinraja** [comment]
Thank you, I'm glad that you find this wrapper useful!

This package uses codemirror 6, so we can use the `@codemirror/lint` to get linting. [Here's a codesandbox example](https://codesandbox.io/s/naughty-bell-qf37f?file=/src/App.tsx) with javascript linting (using eslint). I'll be adding this as an example too.

**Comment by sachinraja** [comment]
Example added [here](https://github.com/sachinraja/rodemirror/tree/main/examples/linting)!

**Comment by nik-swarmagency** [comment]
Great, thanks for the example update I think that will help a lot of people since I eventually found an example by searching the forums for hours lol. See my journey here: https://discuss.codemirror.net/t/cm6-linting-and-error-notifications-highlighting/3425

One last thing that may actually be a bug:
I am trying to have the lintPanel open by default. I found a function (used in basicSetup to bind F8

**Comment by nik-swarmagency** [comment]
^reopened with above comments

**Comment by sachinraja** [comment]
> And if I'm doing something wrong please let me know, as I've been searching through documentation and forums to figure these out.

I can relate, there isn't much information on advanced use cases for codemirror 6. I had to search for about an hour to figure out how to set up linting.

I'm not sure if the `TypeError` there is a codemirror error or an error caused by this package. I checked both the `onEditorViewChange` and `onEditorStateChange` functions and they only seem to be getting called once (using the linting example) when the page is loaded. I'm getting the same error though.

I also was able to reproduce the `openLintPanel` behavior - nothing seems to be happening in the `onEditorViewChange` callback. Though I'm able to reproduce this error outside the library [here](https://codesandbox.io/s/react-17-codemirror-6-forked-2epnq?file=/src/App.js). I think the intended (and working) use is this:
```js
import { keymap } from '@codemirror/view'

const keyBinding: KeyBinding = {
    key: 'Ctrl-Shift-l',
    run: openLintPanel,
  }
  
const keyMapExtension = keymap.of([keyBinding])
  ```

I believe this is also an issue with codemirror. I recommend opening an issue over at [codemirror/codemirror.next](https://github.com/codemirror/codemirror.next/issues).
